dlib/array/array_kernel.h

```diff
@@ -245,10 +245,18 @@ namespace dlib
     {
         try
         {
+            // This is probably the least likely value for a max_size from an old version.
+            // TODO: start version numbering from 1, when it can be reasonably expected that
+            //       there aren't too many objects around any longer serialized without any
+            //       version number (but directly a max_size).
+            const unsigned long version = std::numeric_limits<unsigned long>::max();
+
+            serialize(version,out);
```

Comment:
This breaks backwards compatibility.  Why do you think you need to change the serialization code?

Reply:
For backward compatibiity, I thought. At least when deserializing data from an old version to a new version. But if it's not needed, then that's even better of course. Didn't really have a look at how dlib's serialization works under the hood – if it readily handles the different types, then that's really great.